src/core/qgsmapthemecollection.h

```diff
@@ -150,12 +151,23 @@ class CORE_EXPORT QgsMapThemeCollection : public QObject
          */
         bool hasExpandedStateInfo() const { return mHasExpandedStateInfo; }
 
+        /**
+         * Returns whether information about checked/unchecked state of groups has been recorded
+         * and thus whether checkedGroupNodes() is valid.
+         */
+        bool hasCheckedStateInfo() const { return mHasCheckedStateInfo; }
+
         /**
          * Sets whether the map theme contains valid expanded/collapsed state of nodes
          * \since QGIS 3.2
          */
         void setHasExpandedStateInfo( bool hasInfo ) { mHasExpandedStateInfo = hasInfo; }
 
+        /**
+         * Sets whether the map theme contains valid checked/unchecked state of group nodes
+         */
+        void setHasCheckedStateInfo( bool hasInfo ) { mHasCheckedStateInfo = hasInfo; }
```

Comment:
I'd suggest hiding this from the sip bindings -- it's not something we really want in the public stable API

Reply:
Hi @nyalldawson 
Sorry to come back on such an old comment.

On creating a project with PyQGIS (loading layers, apply QML files etc.) we would like to create the map themes as well. Every bit is supported except this. It's not possible to control the checked state of the group. Why's that? What have been the risks to decide that this should not be in the sip bindings?

Another way to apply map themes to a project would be to create the layer tree completely and then store the current state. But this is much less elegant.

So I would love to enable it in the API. Thanks a lot.